cmd/network/ipv4ll.go

```diff
@@ -0,0 +1,79 @@
+package network
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/j-keck/arping"
+	"github.com/vishvananda/netlink"
+)
+
+func AssignLinkLocalIP(link netlink.Link) error {
+	ifaceName := link.Attrs().Name
+	randSource, err := getPseudoRandomGenerator(ifaceName)
+	if err != nil {
+		return err
+	}
+	// try a random address upto 10 times
+	for i := 0; i < 10; i++ {
+		randGenerator := rand.New(*randSource)
+		randomNum := randGenerator.Uint32()
+		dstIP := getNewIPV4LLAddr(randomNum)
+		_, _, err := arping.PingOverIfaceByName(dstIP, ifaceName)
+		if err != nil {
+			// this ip is not being used
+			addr, err := netlink.ParseAddr(dstIP.String() + "/16")
+			if err != nil {
+				log.Errorf("error while parsing ipv4ll addr, err = %v", err)
+				return err
+			}
+			if err := netlink.AddrAdd(link, addr); err != nil {
+				log.Error("addr add failed")
+				return err
+			}
+			log.Infof("Set %s on %s", dstIP.String(), link.Attrs().Name)
+			return nil
+		}
+	}
+	log.Error("Could not find a suitable ipv4ll")
+	return fmt.Errorf("Could not find a suitable ipv4ll")
+}
+
+func getNewIPV4LLAddr(randomNum uint32) net.IP {
+	byte1 := randomNum & 255 // use least significant 8 bits
```

Comment:
All of 169.254.0.* and 169.254.255.* are not valid:  http://tools.ietf.org/html/rfc3927#section-2.1